email_separator/mail_thread.py

```diff
@@ -44,10 +43,16 @@ def message_route_check_bounce(self, cr, uid, message, context=None):
         email_to = decode_header(message, 'To')
 
         if bounce_alias in email_to:
-            bounce_re = re.compile('%s%s' % (re.escape(bounce_alias),
-                                             re.UNICODE), BOUNCE_EXPR)
-            if bounce_re:
-                message.set_param('To', message['To'].replace('+', '-', 1))
+            bounce_re = re.compile(
+                r'%s\+(\d+)-?([\w.]+)?-?(\d+)?' % re.escape(bounce_alias),
+                re.UNICODE)
+            if bounce_re.search(email_to):
```

Comment:
Please add a small comment pointing to the original code in mass_mail/mail_thread.py.